### PR TITLE
Free esmpy of ` >=8.6.0` pin and pin `iris-grib >=0.20.0`

### LINKED ISSUE #2535: Latest iris-grib v0.20.0 conflicts with ncl and esmpy>=8.6.0
Note: this currently only applies to https://github.com/ESMValGroup/ESMValCore/pull/2178.

See https://app.circleci.com/pipelines/github/ESMValGroup/ESMValCore/11756/workflows/d000ec09-9d09-4cea-b6e1-e4500936f6b0/jobs/49467?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary

This is reproducible with

```bash
mamba create -n test "iris-grib>=0.20.0" ncl "esmpy>=8.6.0"
```

<details>

```bash
error    libmamba Could not solve for environment specs
    The following packages are incompatible
    ├─ esmpy >=8.6.0  is installable with the potential options
    │  ├─ esmpy 8.6.0 would require
    │  │  └─ esmf 8.6.0.*  with the potential options
    │  │     ├─ esmf [8.0.0|8.0.1|...|8.6.1] would require
    │  │     │  ├─ hdf5 [>=1.10.6,<1.10.7.0a0 mpi_openmpi_*|>=1.12.1,<1.12.2.0a0 mpi_openmpi_*], which requires
    │  │     │  │  └─ zlib >=1.2.11,<1.3.0a0 , which can be installed;
    │  │     │  └─ libnetcdf [* mpi_openmpi_*|>=4.7.3,<4.7.3.1.0a0 mpi_openmpi_*|>=4.7.4,<4.7.5.0a0 mpi_openmpi_*|>=4.8.1,<4.8.2.0a0 mpi_openmpi_*|>=4.9.2,<4.9.3.0a0 mpi_openmpi_*] with the potential options
    │  │     │     ├─ libnetcdf 4.9.2 would require
    │  │     │     │  └─ libjpeg-turbo >=2.1.5.1,<3.0a0 , which requires
    │  │     │     │     └─ jpeg <0.0.0a , which can be installed;
    │  │     │     ├─ libnetcdf 4.9.2, which can be installed;
    │  │     │     ├─ libnetcdf [4.7.4|4.8.0|4.8.1] would require
    │  │     │     │  ├─ jpeg >=9d,<10a , which conflicts with any installable versions previously reported;
    │  │     │     │  └─ zlib >=1.2.11,<1.3.0a0 , which can be installed;
    │  │     │     ├─ libnetcdf [4.6.2|4.7.0|...|4.7.4] would require
    │  │     │     │  ├─ jpeg >=9c,<10a , which conflicts with any installable versions previously reported;
    │  │     │     │  └─ zlib >=1.2.11,<1.3.0a0 , which can be installed;
    │  │     │     ├─ libnetcdf 4.8.1 would require
    │  │     │     │  ├─ jpeg >=9e,<10a , which conflicts with any installable versions previously reported;
    │  │     │     │  └─ zlib >=1.2.12,<1.3.0a0 , which can be installed;
    │  │     │     ├─ libnetcdf [4.8.1|4.9.0|4.9.1] would require
    │  │     │     │  └─ jpeg >=9e,<10a , which conflicts with any installable versions previously reported;
    │  │     │     ├─ libnetcdf 4.8.1 would require
    │  │     │     │  ├─ jpeg >=9e,<10a , which conflicts with any installable versions previously reported;
    │  │     │     │  └─ zlib >=1.2.11,<1.3.0a0 , which can be installed;
    │  │     │     └─ libnetcdf 4.9.1 would require
    │  │     │        └─ libjpeg-turbo >=2.1.5.1,<3.0a0 , which can be installed (as previously explained);
    │  │     ├─ esmf [8.1.1|8.2.0|...|8.6.1] would require
    │  │     │  ├─ hdf5 [>=1.10.6,<1.10.7.0a0 |>=1.12.1,<1.12.2.0a0 ], which can be installed (as previously explained);
    │  │     │  └─ libnetcdf [* nompi_*|>=4.8.1,<4.8.2.0a0 ] but there are no viable options
    │  │     │     ├─ libnetcdf 4.9.2 would require
    │  │     │     │  └─ libjpeg-turbo >=2.1.5.1,<3.0a0 , which can be installed (as previously explained);
    │  │     │     ├─ libnetcdf 4.9.2 conflicts with any installable versions previously reported;
    │  │     │     ├─ libnetcdf [4.7.4|4.8.0|4.8.1], which cannot be installed (as previously explained);
    │  │     │     ├─ libnetcdf [4.6.2|4.7.0|...|4.7.4], which cannot be installed (as previously explained);
    │  │     │     ├─ libnetcdf 4.8.1, which cannot be installed (as previously explained);
    │  │     │     ├─ libnetcdf [4.8.1|4.9.0|4.9.1], which cannot be installed (as previously explained);
    │  │     │     ├─ libnetcdf 4.8.1, which cannot be installed (as previously explained);
    │  │     │     └─ libnetcdf 4.9.1, which cannot be installed (as previously explained);
    │  │     └─ esmf [8.6.0|8.6.1] would require
    │  │        └─ libnetcdf [* mpi_mpich_*|>=4.9.2,<4.9.3.0a0 mpi_mpich_*] with the potential options
    │  │           ├─ libnetcdf 4.9.2, which can be installed;
    │  │           ├─ libnetcdf 4.9.2, which can be installed (as previously explained);
    │  │           ├─ libnetcdf [4.7.4|4.8.0|4.8.1], which cannot be installed (as previously explained);
    │  │           ├─ libnetcdf [4.6.2|4.7.0|...|4.7.4], which cannot be installed (as previously explained);
    │  │           ├─ libnetcdf 4.8.1, which cannot be installed (as previously explained);
    │  │           ├─ libnetcdf [4.8.1|4.9.0|4.9.1], which cannot be installed (as previously explained);
    │  │           ├─ libnetcdf 4.8.1, which cannot be installed (as previously explained);
    │  │           └─ libnetcdf 4.9.1, which cannot be installed (as previously explained);
    │  └─ esmpy 8.6.1 would require
    │     └─ esmf 8.6.1.*  with the potential options
    │        ├─ esmf [8.0.0|8.0.1|...|8.6.1], which can be installed (as previously explained);
    │        ├─ esmf [8.1.1|8.2.0|...|8.6.1], which cannot be installed (as previously explained);
    │        ├─ esmf [8.6.0|8.6.1], which cannot be installed (as previously explained);
    │        └─ esmf 8.6.1 would require
    │           ├─ libnetcdf * mpi_openmpi_* with the potential options
    │           │  ├─ libnetcdf 4.9.2, which can be installed (as previously explained);
    │           │  ├─ libnetcdf 4.9.2, which can be installed;
    │           │  ├─ libnetcdf [4.7.4|4.8.0|4.8.1], which cannot be installed (as previously explained);
    │           │  ├─ libnetcdf [4.6.2|4.7.0|...|4.7.4], which cannot be installed (as previously explained);
    │           │  ├─ libnetcdf 4.8.1, which cannot be installed (as previously explained);
    │           │  ├─ libnetcdf [4.8.1|4.9.0|4.9.1], which cannot be installed (as previously explained);
    │           │  ├─ libnetcdf 4.8.1, which cannot be installed (as previously explained);
    │           │  └─ libnetcdf 4.9.1, which cannot be installed (as previously explained);
    │           └─ openmpi >=5.0.5,<6.0a0 , which requires
    │              └─ libzlib >=1.3.1,<2.0a0  but there are no viable options
    │                 ├─ libzlib 1.3.1 would require
    │                 │  └─ zlib 1.3.1 *_1, which conflicts with any installable versions previously reported;
    │                 └─ libzlib 1.3.1 would require
    │                    └─ zlib 1.3.1 *_0, which conflicts with any installable versions previously reported;
    ├─ iris-grib >=0.20.0  is not installable because it requires
    │  ├─ eccodes >=2.33  but there are no viable options
    │  │  ├─ eccodes [2.33.0|2.34.0|2.34.1] would require
    │  │  │  ├─ hdf5 >=1.14.3,<1.14.4.0a0 , which conflicts with any installable versions previously reported;
    │  │  │  ├─ jasper [>=4.1.0,<5.0a0 |>=4.2.0,<5.0a0 |>=4.2.1,<5.0a0 ], which requires
    │  │  │  │  └─ libjpeg-turbo >=3.0.0,<4.0a0 , which requires
    │  │  │  │     └─ jpeg <0.0.0a , which can be installed;
    │  │  │  └─ libnetcdf >=4.9.2,<4.9.3.0a0  with the potential options
    │  │  │     ├─ libnetcdf 4.9.2, which can be installed;
    │  │  │     ├─ libnetcdf 4.9.2, which can be installed (as previously explained);
    │  │  │     ├─ libnetcdf 4.9.2, which can be installed;
    │  │  │     ├─ libnetcdf 4.9.2, which cannot be installed (as previously explained);
    │  │  │     └─ libnetcdf 4.9.2 conflicts with any installable versions previously reported;
    │  │  └─ eccodes [2.35.0|2.35.1|2.36.0|2.37.0|2.38.0] would require
    │  │     ├─ jasper [>=4.2.3,<5.0a0 |>=4.2.4,<5.0a0 ], which cannot be installed (as previously explained);
    │  │     └─ libnetcdf >=4.9.2,<4.9.3.0a0  with the potential options
    │  │        ├─ libnetcdf 4.9.2, which can be installed;
    │  │        ├─ libnetcdf 4.9.2, which can be installed (as previously explained);
    │  │        ├─ libnetcdf 4.9.2, which can be installed;
    │  │        ├─ libnetcdf 4.9.2, which cannot be installed (as previously explained);
    │  │        └─ libnetcdf 4.9.2 conflicts with any installable versions previously reported;
    │  └─ iris >=3.0.2  with the potential options
    │     ├─ iris [3.0.2|3.0.3|3.0.4|3.1.0] would require
    │     │  └─ cartopy >=0.18  with the potential options
    │     │     ├─ cartopy [0.18.0|0.19.0|0.19.0.post1] would require
    │     │     │  └─ proj >=7.2.0,<7.2.1.0a0 , which can be installed;
    │     │     ├─ cartopy 0.18.0 would require
    │     │     │  └─ proj >=7.1.1,<7.1.2.0a0 , which can be installed;
    │     │     ├─ cartopy 0.18.0 would require
    │     │     │  └─ proj >=7.1.0,<7.1.1.0a0 , which can be installed;
    │     │     ├─ cartopy [0.20.0|0.20.1] would require
    │     │     │  └─ proj >=8.1.1,<8.1.2.0a0 , which can be installed;
    │     │     ├─ cartopy 0.20.0 would require
    │     │     │  └─ proj >=8.0.1,<8.0.2.0a0 , which can be installed;
    │     │     ├─ cartopy 0.20.0 would require
    │     │     │  └─ proj >=8.1.0,<8.1.1.0a0 , which can be installed;
    │     │     ├─ cartopy [0.20.1|0.20.2] would require
    │     │     │  └─ proj >=8.2.0,<8.2.1.0a0 , which can be installed;
    │     │     ├─ cartopy 0.20.2 would require
    │     │     │  └─ proj >=8.2.1,<8.2.2.0a0 , which can be installed;
    │     │     ├─ cartopy [0.20.2|0.20.3] would require
    │     │     │  └─ proj >=9.0.1,<9.0.2.0a0 , which can be installed;
    │     │     ├─ cartopy 0.20.2 would require
    │     │     │  └─ proj >=9.0.0,<9.0.1.0a0 , which can be installed;
    │     │     ├─ cartopy [0.21.0|0.21.1|0.22.0] would require
    │     │     │  └─ pyproj [>=3.0.0 |>=3.1.0 ] with the potential options
    │     │     │     ├─ pyproj [3.1.0|3.2.0|3.2.1|3.3.0|3.3.1] would require
    │     │     │     │  └─ proj >=8.0.1,<8.0.2.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.0.0|3.0.0.post1|3.0.1|3.1.0] would require
    │     │     │     │  └─ proj >=7.2.0,<7.2.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.1.0|3.2.0|3.2.1] would require
    │     │     │     │  └─ proj >=8.1.0,<8.1.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.0.1|3.1.0] would require
    │     │     │     │  └─ proj >=8.0.0,<8.0.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.2.1|3.3.0] would require
    │     │     │     │  └─ proj >=8.2.0,<8.2.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj 3.2.1 would require
    │     │     │     │  └─ proj >=8.1.1,<8.1.2.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.2.1|3.3.0|3.3.1] would require
    │     │     │     │  └─ proj >=9.0.0,<9.0.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj 3.3.0 would require
    │     │     │     │  └─ proj >=8.2.1,<8.2.2.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.3.1|3.4.0] would require
    │     │     │     │  └─ proj >=9.0.1,<9.0.2.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.4.0|3.4.1] would require
    │     │     │     │  └─ proj >=9.1.0,<9.1.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.4.1|3.5.0] would require
    │     │     │     │  └─ proj >=9.1.1,<9.1.2.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.5.0|3.6.0] would require
    │     │     │     │  └─ proj >=9.2.0,<9.2.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj [3.6.0|3.6.1] would require
    │     │     │     │  └─ proj >=9.2.1,<9.2.2.0a0 , which can be installed;
    │     │     │     ├─ pyproj 3.6.1 would require
    │     │     │     │  └─ proj >=9.5.0,<9.6.0a0 , which can be installed;
    │     │     │     ├─ pyproj 3.6.1 would require
    │     │     │     │  └─ proj >=9.4.1,<9.5.0a0 , which can be installed;
    │     │     │     ├─ pyproj 3.6.1 would require
    │     │     │     │  └─ proj >=9.3.0,<9.3.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj 3.6.1 would require
    │     │     │     │  └─ proj >=9.4.0,<9.4.1.0a0 , which can be installed;
    │     │     │     ├─ pyproj 3.6.1 would require
    │     │     │     │  └─ proj >=9.3.1,<9.3.2.0a0 , which conflicts with any installable versions previously reported;
    │     │     │     ├─ pyproj 3.6.1 would require
    │     │     │     │  └─ proj >=9.4.0,<9.5.0a0 , which can be installed;
    │     │     │     └─ pyproj 3.6.1 would require
    │     │     │        └─ python [>=3.13.0rc1,<3.14.0a0 |>=3.13.0rc2,<3.14.0a0 ], which requires
    │     │     │           └─ _python_rc, which does not exist (perhaps a missing channel);
    │     │     ├─ cartopy 0.23.0 would require
    │     │     │  └─ pyproj >=3.3.1  with the potential options
    │     │     │     ├─ pyproj [3.1.0|3.2.0|3.2.1|3.3.0|3.3.1], which can be installed (as previously explained);
    │     │     │     ├─ pyproj [3.2.1|3.3.0|3.3.1], which can be installed (as previously explained);
    │     │     │     ├─ pyproj [3.3.1|3.4.0], which can be installed (as previously explained);
    │     │     │     ├─ pyproj [3.4.0|3.4.1], which can be installed (as previously explained);
    │     │     │     ├─ pyproj [3.4.1|3.5.0], which can be installed (as previously explained);
    │     │     │     ├─ pyproj [3.5.0|3.6.0], which can be installed (as previously explained);
    │     │     │     ├─ pyproj [3.6.0|3.6.1], which can be installed (as previously explained);
    │     │     │     ├─ pyproj 3.6.1, which can be installed (as previously explained);
    │     │     │     ├─ pyproj 3.6.1, which can be installed (as previously explained);
    │     │     │     ├─ pyproj 3.6.1, which can be installed (as previously explained);
    │     │     │     ├─ pyproj 3.6.1, which can be installed (as previously explained);
    │     │     │     ├─ pyproj 3.6.1, which cannot be installed (as previously explained);
    │     │     │     ├─ pyproj 3.6.1, which can be installed (as previously explained);
    │     │     │     └─ pyproj 3.6.1, which cannot be installed (as previously explained);
    │     │     └─ cartopy 0.23.0 would require
    │     │        └─ python >=3.13.0rc2,<3.14.0a0 , which cannot be installed (as previously explained);
    │     └─ iris [3.10.0|3.2.0|...|3.9.0] would require
    │        └─ cartopy [>=0.20 |>=0.21 ] with the potential options
    │           ├─ cartopy [0.20.0|0.20.1], which can be installed (as previously explained);
    │           ├─ cartopy 0.20.0, which can be installed (as previously explained);
    │           ├─ cartopy 0.20.0, which can be installed (as previously explained);
    │           ├─ cartopy [0.20.1|0.20.2], which can be installed (as previously explained);
    │           ├─ cartopy 0.20.2, which can be installed (as previously explained);
    │           ├─ cartopy [0.20.2|0.20.3], which can be installed (as previously explained);
    │           ├─ cartopy 0.20.2, which can be installed (as previously explained);
    │           ├─ cartopy [0.21.0|0.21.1|0.22.0], which can be installed (as previously explained);
    │           ├─ cartopy 0.23.0, which can be installed (as previously explained);
    │           └─ cartopy 0.23.0, which cannot be installed (as previously explained);
    └─ ncl is installable with the potential options
       ├─ ncl [6.5.0|6.6.0|6.6.2] would require
       │  ├─ hdf5 >=1.10.4,<1.10.5.0a0 , which can be installed;
       │  └─ jasper >=1.900.1,<2.0a0  but there are no viable options
       │     ├─ jasper 1.900.1 conflicts with any installable versions previously reported;
       │     └─ jasper 1.900.1 would require
       │        └─ jpeg 9b , which does not exist (perhaps a missing channel);
       ├─ ncl 6.3.0 would require
       │  └─ gdal 2.1.* , which does not exist (perhaps a missing channel);
       ├─ ncl 6.4.0 would require
       │  └─ libgdal 2.2.* , which does not exist (perhaps a missing channel);
       ├─ ncl [6.4.0|6.5.0] would require
       │  └─ libgdal >=2.2.4,<2.3.0a0 , which does not exist (perhaps a missing channel);
       ├─ ncl 6.6.2 would require
       │  ├─ hdf5 >=1.10.6,<1.10.7.0a0 , which can be installed (as previously explained);
       │  └─ jasper >=1.900.1,<2.0a0  but there are no viable options
       │     ├─ jasper 1.900.1 conflicts with any installable versions previously reported;
       │     └─ jasper 1.900.1, which cannot be installed (as previously explained);
       ├─ ncl 6.6.2 would require
       │  ├─ hdf5 >=1.10.5,<1.10.6.0a0 , which can be installed;
       │  └─ jasper >=1.900.1,<2.0a0  but there are no viable options
       │     ├─ jasper 1.900.1 conflicts with any installable versions previously reported;
       │     └─ jasper 1.900.1, which cannot be installed (as previously explained);
       ├─ ncl 6.6.2 would require
       │  └─ jasper >=2.0.33,<3.0a0 , which conflicts with any installable versions previously reported;
       ├─ ncl 6.6.2 would require
       │  ├─ jasper >=1.900.1,<2.0a0  but there are no viable options
       │  │  ├─ jasper 1.900.1 conflicts with any installable versions previously reported;
       │  │  └─ jasper 1.900.1, which cannot be installed (as previously explained);
       │  └─ libnetcdf >=4.7.4,<4.7.5.0a0 , which cannot be installed (as previously explained);
       ├─ ncl 6.6.2 would require
       │  └─ proj >=6.3.0,<6.3.1.0a0 , which conflicts with any installable versions previously reported;
       ├─ ncl 6.6.2 would require
       │  └─ proj >=6.2.0,<6.2.1.0a0 , which conflicts with any installable versions previously reported;
       ├─ ncl 6.6.2 would require
       │  ├─ jasper >=2.0.14,<3.0a0 , which conflicts with any installable versions previously reported;
       │  └─ libnetcdf >=4.8.1,<4.8.2.0a0 , which cannot be installed (as previously explained);
       ├─ ncl 6.6.2 would require
       │  └─ proj >=6.2.1,<6.2.2.0a0 , which conflicts with any installable versions previously reported;
       ├─ ncl 6.6.2 would require
       │  ├─ jasper >=1.900.1,<2.0a0  but there are no viable options
       │  │  ├─ jasper 1.900.1 conflicts with any installable versions previously reported;
       │  │  └─ jasper 1.900.1, which cannot be installed (as previously explained);
       │  └─ libnetcdf >=4.8.1,<4.8.2.0a0 , which cannot be installed (as previously explained);
       ├─ ncl 6.6.2 would require
       │  ├─ hdfeos2 with the potential options
       │  │  ├─ hdfeos2 2.19.1 would require
       │  │  │  ├─ jpeg 9* , which conflicts with any installable versions previously reported;
       │  │  │  └─ zlib [1.2* |1.2.* ], which can be installed;
       │  │  ├─ hdfeos2 2.19.1 would require
       │  │  │  ├─ jpeg 9* , which conflicts with any installable versions previously reported;
       │  │  │  └─ zlib 1.2.11 , which can be installed;
       │  │  ├─ hdfeos2 2.20 would require
       │  │  │  ├─ jpeg >=9c,<10a , which conflicts with any installable versions previously reported;
       │  │  │  └─ zlib >=1.2.11,<1.3.0a0 , which can be installed;
       │  │  ├─ hdfeos2 2.20 would require
       │  │  │  ├─ jpeg >=9e,<10a , which conflicts with any installable versions previously reported;
       │  │  │  └─ zlib >=1.2.11,<1.3.0a0 , which can be installed;
       │  │  ├─ hdfeos2 2.20 would require
       │  │  │  └─ libjpeg-turbo [>=2.1.5.1,<3.0a0 |>=3.0.0,<4.0a0 ] with the potential options
       │  │  │     ├─ libjpeg-turbo 2.1.5.1, which can be installed (as previously explained);
       │  │  │     └─ libjpeg-turbo 3.0.0, which cannot be installed (as previously explained);
       │  │  └─ hdfeos2 2.20 would require
       │  │     └─ jpeg >=9e,<10a , which conflicts with any installable versions previously reported;
       │  └─ libjpeg-turbo >=2.1.5.1,<3.0a0 , which can be installed (as previously explained);
       ├─ ncl 6.6.2 would require
       │  ├─ esmf <8.6.0  with the potential options
       │  │  ├─ esmf [7.1.0|7.1.0r|8.0.0|8.0.1|8.1.0] would require
       │  │  │  └─ libnetcdf [>=4.6.2,<4.6.3.0a0 |>=4.6.2,<4.7.0a0 |...|>=4.7.4,<4.7.5.0a0 mpi_openmpi_*], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.0.0|8.0.1|...|8.4.2] would require
       │  │  │  ├─ hdf5 [>=1.10.6,<1.10.7.0a0 mpi_mpich_*|>=1.12.1,<1.12.2.0a0 mpi_mpich_*], which can be installed (as previously explained);
       │  │  │  └─ libnetcdf [* mpi_mpich_*|>=4.7.3,<4.7.3.1.0a0 mpi_mpich_*|>=4.7.4,<4.7.5.0a0 mpi_mpich_*|>=4.8.1,<4.8.2.0a0 mpi_mpich_*|>=4.9.2,<4.9.3.0a0 mpi_mpich_*] with the potential options
       │  │  │     ├─ libnetcdf 4.9.2, which can be installed;
       │  │  │     ├─ libnetcdf 4.9.2, which can be installed (as previously explained);
       │  │  │     ├─ libnetcdf [4.7.4|4.8.0|4.8.1], which cannot be installed (as previously explained);
       │  │  │     ├─ libnetcdf [4.6.2|4.7.0|...|4.7.4], which cannot be installed (as previously explained);
       │  │  │     ├─ libnetcdf 4.8.1, which cannot be installed (as previously explained);
       │  │  │     ├─ libnetcdf [4.8.1|4.9.0|4.9.1], which cannot be installed (as previously explained);
       │  │  │     ├─ libnetcdf 4.8.1, which cannot be installed (as previously explained);
       │  │  │     └─ libnetcdf 4.9.1, which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.0.0|8.0.1|...|8.6.1], which can be installed (as previously explained);
       │  │  ├─ esmf [8.1.0|8.1.1] would require
       │  │  │  └─ libnetcdf [>=4.8.0,<4.8.1.0a0 |>=4.8.0,<4.8.1.0a0 mpi_mpich_*|>=4.8.0,<4.8.1.0a0 mpi_openmpi_*], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.1.1|8.2.0|...|8.6.1], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.4.0|8.4.1] would require
       │  │  │  └─ libnetcdf [>=4.9.1,<4.9.2.0a0 |>=4.9.1,<4.9.2.0a0 mpi_mpich_*|>=4.9.1,<4.9.2.0a0 mpi_openmpi_*], which cannot be installed (as previously explained);
       │  │  └─ esmf 8.4.2 would require
       │  │     └─ libnetcdf >=4.9.2,<4.9.3.0a0 mpi_mpich_* with the potential options
       │  │        ├─ libnetcdf 4.9.2, which can be installed;
       │  │        └─ libnetcdf 4.9.2, which can be installed (as previously explained);
       │  └─ libjpeg-turbo >=3.0.0,<4.0a0 , which cannot be installed (as previously explained);
       ├─ ncl 6.6.2 would require
       │  ├─ esmf <8.6.0  with the potential options
       │  │  ├─ esmf [7.1.0|7.1.0r|8.0.0|8.0.1|8.1.0], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.0.0|8.0.1|...|8.4.2], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.0.0|8.0.1|...|8.6.1], which can be installed (as previously explained);
       │  │  ├─ esmf [8.1.0|8.1.1], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.1.1|8.2.0|...|8.6.1], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.4.0|8.4.1], which cannot be installed (as previously explained);
       │  │  └─ esmf 8.4.2, which cannot be installed (as previously explained);
       │  ├─ hdfeos2 with the potential options
       │  │  ├─ hdfeos2 2.19.1, which cannot be installed (as previously explained);
       │  │  ├─ hdfeos2 2.19.1, which cannot be installed (as previously explained);
       │  │  ├─ hdfeos2 2.20, which cannot be installed (as previously explained);
       │  │  ├─ hdfeos2 2.20, which cannot be installed (as previously explained);
       │  │  ├─ hdfeos2 2.20, which can be installed (as previously explained);
       │  │  └─ hdfeos2 2.20, which cannot be installed (as previously explained);
       │  ├─ jasper >=4.2.4,<5.0a0 , which cannot be installed (as previously explained);
       │  ├─ libzlib >=1.3.1,<2.0a0 , which cannot be installed (as previously explained);
       │  └─ zlib with the potential options
       │     ├─ zlib 1.2.11, which can be installed;
       │     ├─ zlib [1.2.12|1.2.13], which can be installed;
       │     ├─ zlib [1.3|1.3.1] conflicts with any installable versions previously reported;
       │     ├─ zlib 1.2.8, which can be installed;
       │     └─ zlib 1.3.1 would require
       │        └─ libzlib 1.3.1 h4ab18f5_1, which cannot be installed (as previously explained);
       ├─ ncl 6.6.2 would require
       │  ├─ esmf <8.6.0  with the potential options
       │  │  ├─ esmf [7.1.0|7.1.0r|8.0.0|8.0.1|8.1.0], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.0.0|8.0.1|...|8.4.2], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.0.0|8.0.1|...|8.6.1], which can be installed (as previously explained);
       │  │  ├─ esmf [8.1.0|8.1.1], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.1.1|8.2.0|...|8.6.1], which cannot be installed (as previously explained);
       │  │  ├─ esmf [8.4.0|8.4.1], which cannot be installed (as previously explained);
       │  │  └─ esmf 8.4.2, which cannot be installed (as previously explained);
       │  ├─ libgdal >=3.8.4,<3.9.0a0  with the potential options
       │  │  ├─ libgdal 3.8.4 would require
       │  │  │  └─ poppler [>=24.2.0,<24.3.0a0 |>=24.3.0,<24.4.0a0 ], which requires
       │  │  │     └─ lcms2 >=2.16,<3.0a0 , which requires
       │  │  │        └─ libjpeg-turbo >=3.0.0,<4.0a0 , which cannot be installed (as previously explained);
       │  │  ├─ libgdal [3.8.4|3.8.5] would require
       │  │  │  └─ proj >=9.4.0,<9.4.1.0a0 , which can be installed;
       │  │  ├─ libgdal 3.8.5 would require
       │  │  │  └─ proj >=9.4.1,<9.5.0a0 , which can be installed;
       │  │  └─ libgdal 3.8.5 would require
       │  │     └─ proj >=9.4.0,<9.5.0a0 , which can be installed;
       │  └─ proj >=9.3.1,<9.3.2.0a0 , which conflicts with any installable versions previously reported;
       ├─ ncl 6.6.2 would require
       │  └─ libnetcdf >=4.9.1,<4.9.2.0a0 , which cannot be installed (as previously explained);
       └─ ncl 6.6.2 would require
          └─ libjpeg-turbo >=2.1.5.1,<3.0a0 , which can be installed (as previously explained).
critical libmamba Could not solve for environment specs
```

</details>

We need the latest iris-grib version (0.20.0) due to a serious bug that prevents opening any file (https://github.com/SciTools/iris-grib/issues/520).

@ESMValGroup/technical-lead-development-team any suggestion what we could do here? I really fear that this cannot be solved due to NCL issues 😩

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

Closes #2535 

Long story short:

- `iris-grib` needs be pinned due to #2178 
- `ncl` has a hard pin `esmf<8.6.0` so whatever `ncl` gets installed for `esmf/esmpy>=8.6.0` is about two years old (still 6.6.2 but a very old build, that did not have the pin in)
- it is rather safe to use `esmf/esmpy<8.6.0` ie `8.4.2` as per https://github.com/SciTools/iris-esmf-regrid/issues/415

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by valeriupredoi** [comment]
(...) = ncl build h7cb714c_54 which the latest build :relaxed:

File: environment.yml [hunk 1]
```diff
@@ -12,15 +12,15 @@ dependencies:
   - dask-jobqueue
   - distributed
   - esgf-pyclient >=0.3.1
-  - esmpy >=8.6.0  # github.com/SciTools-incubator/iris-esmf-regrid/pull/342
+  - esmpy
```
Comment (schlunma) [inline environment.yml:15]:
```suggestion
  - esmpy !=8.1.0
```

As it [used to be](https://github.com/ESMValGroup/ESMValCore/pull/2402/files#diff-9efd195f4e9bfb79ccd456a1d8370fafcc4bcb0b00ea3799222667d2ae818533L16)?

It's also the same pin as in `setup.py`.

Comment (valeriupredoi) [reply environment.yml:15]: good call!

**Comment by valeriupredoi** [comment]
@schlunma cheers Manu, go ahead and merge at will :beer:

**Review by schlunma — APPROVED** [review]
Great, thanks V!!

**Pull request merged**